tests/test_interface.py

```diff
@@ -529,11 +530,26 @@ def test_ecdsa(self):
 
 
     # TEST: Generation errors
+    for idx, (kwargs, err_msg) in enumerate([
+        # Error on empty password
+        ({"password": ""},
+          "encryption password must be 1 or more characters long"),
+        # Error on 'password' and 'prompt=True'
+        ({"password": pw, "prompt": True},
+          "passing 'password' and 'prompt=True' is not allowed")]):
```

Comment:
I agree that it's annoying and prone to break. As a matter of fact I'm pretty sure that it will break in the next build... I just skimmed through the latest `cryptography` release and saw that they changed an error message that we check for here.

That said, I think there is merit in more granular error test assertions that go beyond the type. I've seen tests in the past, which meant to check for one error and passed although a different error was raised, because they had the same type.

So I think it's a fair trade-off to update the expected error strings in tests every now and then, if it gives us stronger tests. And once we have a more solid exception taxonomy (see #191), I'm happy to remove the message assertions. What do you think?

Reply:
FYI: Looks like I was wrong about the [`cryptography` update](https://github.com/pyca/cryptography/commit/58494b41d6ecb0f56b7c5f05d5f5e3ca0320d494#diff-97598491194e5ffd150e82dea4ddb874833c27432477fada5ef1bc9990c4b70fL138-R133) breaking our build. That particular error message is raised by ourselves.

https://github.com/secure-systems-lab/securesystemslib/blob/d05dc6a42908b9ef3d86e8074df6cf0b8e8280b3/securesystemslib/rsa_keys.py#L1062

So no need for action for now... :)